clowder-framework/clowder2-frontend
# Add ESLintPlugin to webpack dev build

## Problem
ESLint does not run as a part of the webpack dev build defined in `webpack.config.dev.js`

Starts to fix #22 but does not yet involve `prettier`

## Approach
* Added ESLint Webpack plugin, which runs ESLint as part of the Webpack dev build - this runs ESLint as a part of your IDE's webpack dev bundling
* Added some new ESLint plugins/presets for typescript and react (these may not be needed, but supposedly offer some best practices)
* Explicitly exclude `node_modules` / `dist` / `build` folders in several places, for sanity's sake
* Turned down `.eslintrc` in some places from `error` to `warn` - this is temporary to avoid breaking the build, but we can turn these up as we work through resolving the existing warnings

## How to Test
1. Checkout and run this branch locally
2. Run eslint manually from command line: `npm run lint`
    * This should report many warnings, but 0 errors
3. Run the webpack dev build from your IDE or command-line: `npm start`
    * NOTE: the `ESLIntPlugin` for Webpack automatically runs eslint in this case
    * This should report many warnings, but 0 errors
4. Run the webpack prod build using the command-line: `npm run build` (used by Docker)
    * NOTE: the `preinstall` hook automatically runs eslint in this case
    * This should report many warnings, but 0 errors
